### PR TITLE
SQL dump tool

### LINKED ISSUE #60: Ability to invoke databasedump.sh with parameters
Right now it's possible, to write username and password of the db-user into the [mysql backup script ](https://github.com/kinafu/borg-cron-helper/blob/master/tools/databasedump.sh)directly.
The ability to have this kind of interface
`$ ./databasedump.sh backup password /root/backup/db`
additionally to
```
USER="backup"
PASSWORD="password"
DIR="/root/backup/db"
```
would help in avoiding redundancy and facilitate maintainability.

Right now the easiest way is to create copies of `databasedump.sh` with "hard-coded" credentials.
If one needs to make a change in the code executing the backup, he would need to update all of his copies.

### PR DESCRIPTION
**Pull request by kinafu**
Fixes #60 
- several ways to pass arguments (cli parameters, env variables, hard-coded variables within the script file)

### CODE REVIEW

File: tools/databasedump.sh [hunk 1]
```diff
@@ -1,28 +1,115 @@
 #!/bin/sh
-# Dumps each database separately. Execute this (as another user with more privileges)
+# Dumps each database separately into a single file. Execute this (as another user with more privileges)
 # before the backup is executed.
-# thanks https://stackoverflow.com/questions/9497869/export-and-import-all-mysql-databases-at-one-time#answer-26096339, modified
+# thanks https://stackoverflow.com/questions/9497869/export-and-import-all-mysql-DATABASES-at-one-time#answer-26096339, modified
```
Comment (rugk) [inline tools/databasedump.sh]: ???

File: tools/databasedump.sh [hunk 2]
```diff
@@ -1,28 +1,115 @@
...
 	fi
 
-	echo "Dumping database: $db"
-	# (variable intentionally not quoted as it is contains custom directives)
+	info_log "MySQL Backup dumping database: \"$db\""
+	# (specialParams intentionally not quoted as it is contains custom directives)
 	# shellcheck disable=SC2086
-	mysqldump -u $USER $PASSWORD $specialParams --databases "$db" > "$DIR/$db.sql"
+	mysqldump -u "$USER" "$PASSWORD" $specialParams --databases "$db" > "$DESTINATION/$db.sql"
+	exitcode_dump=$?
```
Comment (rugk) [inline tools/databasedump.sh]: Please use camelCase for variables.

**Review by rugk — CHANGES_REQUESTED** [review]
More than I thought the resolution to this issue would change, but this is fine.

Only some small nitpicks…

File: tools/databasedump.sh (hunk 1 above)
Comment (kinafu) [reply tools/databasedump.sh]: :dancer:

**Review by rugk — APPROVED** [review]

**Pull request merged**